Commons/src/main/java/tw/waterball/judgegirl/commons/utils/StreamUtils.java

```diff
@@ -114,6 +109,27 @@
         return zipToList(map.entrySet(), zipAndMap);
     }
 
+    public static <T, L, R> List<T> zipToList(L[] left, R[] right,
+                                              BiFunction<L, R, T> zipAndMap) {
+        return range(0, left.length)
+                .mapToObj(i -> zipAndMap.apply(left[i], right[i]))
+                .collect(toList());
+    }
+
+    public static <T, L, R> List<T> zipToList(Collection<L> left, Collection<R> right,
+                                              BiPredicate<L, R> matching,
+                                              BiFunction<L, R, T> zipAndMap) {
+        List<T> list = new ArrayList<>();
```

Comment:
var list = new LinkedList<>();

Reply:
ArrayList will be more efficient as we have known its size